conda/plugins/shells/posix/posix_activate.py

```diff
@@ -0,0 +1,224 @@
+        Set context.dev if a --dev flag exists.
+        For activate, set self.env_name_or_prefix and self.stack.
+        """
+        self.command = args.command
+        context.dev = args.dev or context.dev
+
+        if self.command == "activate":
+            self.env_name_or_prefix = args.env or "base"
+
+            if args.stack is None:
+                self.stack = context.auto_stack and context.shlvl <= context.auto_stack
+            else:
+                self.stack = args.stack
+
+        return
+
+
+def get_parsed_args(argv: list[str]) -> argparse.Namespace:
+    """
+    Parse CLI arguments to determine desired command.
+    Create namespace with 'command' and 'env' keys.
+    """
+    parser = argparse.ArgumentParser(
+        "posix_plugin_current_logic",
+        description="Process conda activate, deactivate, and reactivate",
+    )
+
+    commands = parser.add_subparsers(
+        required=True,
+        dest="command",
+    )
+
+    activate = commands.add_parser(
+        "activate",
+        help="Activate a conda environment",
+    )
+    activate.add_argument(
+        "env",
+        metavar="env_name_or_prefix",
+        default=None,
+        type=str,
+        nargs="?",
+        help="""
+            The environment name or prefix to activate. If the prefix is a relative path,
+            it must start with './' (or '.\' on Windows). If no environment is specified,
+            the base environment will be activated.
+            """,
+    )
+    stack = activate.add_mutually_exclusive_group()
+    stack.add_argument(
+        "--stack",
+        action="store_true",
+        help="""
+        Stack the environment being activated on top of the
+        previous active environment, rather replacing the
+        current active environment with a new one. Currently,
+        only the PATH environment variable is stacked. This
+        may be enabled implicitly by the 'auto_stack'
+        configuration variable.
+        """,
+    )
+    stack.add_argument(
+        "--no-stack",
+        dest="stack",
+        action="store_false",
+        help="Do not stack the environment. Overrides 'auto_stack' setting.",
+    )
+    activate.add_argument(
+        "--dev", action="store_true", default=False, help=argparse.SUPPRESS
+    )
+
+    deactivate = commands.add_parser(
+        "deactivate", help="Deactivate the current active conda environment"
+    )
+    deactivate.add_argument(
+        "--dev", action="store_true", default=False, help=argparse.SUPPRESS
+    )
+
+    reactivate = commands.add_parser(
+        "reactivate",
+        help="Reactivate the current conda environment, updating environment variables",
+    )
+    reactivate.add_argument(
+        "--dev", action="store_true", default=False, help=argparse.SUPPRESS
+    )
+
+    try:
+        args = parser.parse_args(argv)
+    except BaseException:
+        # avoid evaluation of help strings
```

Comment:
we want to keep exceptions targeted so we don't end up catching too much:

```suggestion
    except SystemExit:
        # SystemExit: help blurb was printed, intercepting SystemExit(0) to avoid
        # evaluation of help strings by shell interface
```